### ISSUE TITLE
CookieError Ilegal key %r, Jupyterhub not sanitizing header keys properly

### ISSUE THREAD

**Issue by atnjqt** [jupyterhub/jupyterhub#4330]
### Bug description

End users occasionally are receiving timeouts on **Jupyterhub 3.1.0**. On review, the error log `/var/log/jupyterhub.log` contains many unexpected references to an Illegal Key in HTTPCookie.

#### Expected behaviour

End users running R kernel ipynbs should not get unexpected timeouts due to invalid key in HTTP Cookies when authenticating to the hub via GitHub OAuth.

#### Actual behaviour

End users occasionally get timeouts. Below is a reference of the actual error log, the first hit. Interestingly the error log has thousands of hits for that same illegal key value, which I believe is a Google Analytics tag & measurement ID.

```python
[E 2023-01-24 00:12:16.024 JupyterHub web:1798] Uncaught exception GET / (::ffff:xxx.xxx.xxx.xxx)
    HTTPServerRequest(protocol='https', host='commjhub.asc.upenn.edu', method='GET', uri='/', version='HTTP/1.1', remote_ip='::ffff:xxx.xxx.xxx.xxx')
    Traceback (most recent call last):
      File "/usr/local/anaconda3/lib/python3.8/site-packages/tornado/web.py", line 1711, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/usr/local/anaconda3/lib/python3.8/site-packages/jupyterhub/handlers/base.py", line 1391, in get
        self.redirect(url_path_join(self.hub.base_url, path), permanent=False)
      File "/usr/local/anaconda3/lib/python3.8/site-packages/tornado/web.py", line 816, in redirect
        self.finish()
      File "/usr/local/anaconda3/lib/python3.8/site-packages/jupyterhub/handlers/base.py", line 184, in finish
        super().finish(*args, **kwargs)
      File "/usr/local/anaconda3/lib/python3.8/site-packages/tornado/web.py", line 1161, in finish
        self._log()
      File "/usr/local/anaconda3/lib/python3.8/site-packages/tornado/web.py", line 1746, in _log
        self.application.log_request(self)
      File "/usr/local/anaconda3/lib/python3.8/site-packages/tornado/web.py", line 2262, in log_request
        self.settings["log_function"](handler)
      File "/usr/local/anaconda3/lib/python3.8/site-packages/jupyterhub/log.py", line 143, in log_request
        headers = _scrub_headers(request.headers)
      File "/usr/local/anaconda3/lib/python3.8/site-packages/jupyterhub/log.py", line 108, in _scrub_headers
        c = SimpleCookie(headers['Cookie'])
      File "/usr/local/anaconda3/lib/python3.8/http/cookies.py", line 480, in __init__
        self.load(input)
      File "/usr/local/anaconda3/lib/python3.8/http/cookies.py", line 529, in load
        self.__parse_string(rawdata)
      File "/usr/local/anaconda3/lib/python3.8/http/cookies.py", line 593, in __parse_string
        self.__set(key, rval, cval)
      File "/usr/local/anaconda3/lib/python3.8/http/cookies.py", line 485, in __set
        M.set(key, real_value, coded_value)
      File "/usr/local/anaconda3/lib/python3.8/http/cookies.py", line 352, in set
        raise CookieError('Illegal key %r' % (key,))
    http.cookies.CookieError: Illegal key '_ga_SR2P22QVQQ,G-0HYE8YG0M6'
```

### How to reproduce

Pass an invalid key for HTTP cookies. In my case, I think it's the comma value in the GA `http.cookies.CookieError: Illegal key '_ga_SR2P22QVQQ,G-0HYE8YG0M6'` based on information here:  https://docs.python.org/3/library/http.cookies.html#module-http.cookies
> Note: On encountering an invalid cookie, CookieError is raised, so if your cookie data comes from a browser you should always prepare for invalid data and catch CookieError on parsing.

### Your personal set up

 - OS: Ubuntu 20.04.5 LTS
 
 - Version(s): 
   - Jupyterhub 3.1.0
   - Python 3.8.8

**Comment by atnjqt** [comment]
I believe the error is reproducible by manually adding a cookie to your browser session with an invalid char:

1. Right click **Inspect > Console** and add something with an invalid char from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes
```console
document.cookie="_ga_EtienneIsACookieMonster,breakingnow=G-blaaaaahblahblah"
```
2. Confirm malformatted cookie is in **Inspect > Application > Cookies** 
3. Try and run a notebook with Python or R kernel
4. Observe the Kernel shows _Unknown_ and the cell does not run
5. Delete the malformatted cookie and rerun cell

![image](https://user-images.githubusercontent.com/44907911/218152480-a15cba23-e18c-49a3-8456-fd3e0979e449.png)

It might just be the case that Jupyterhub isn't sanitizing HTTPCookies which is unexpected and arguably a bug depending on your perspective.. In my case, I think end users were respectively accessing _some_ publicly accessible site that has a malformatted GA4 tag (the migration from UA to GA4 is ongoing for 2023, so it makes sense this just came up randomly)

**Comment by atnjqt** [comment]
Searching back I did see some related issues... there was a commit here that I thought is maybe relevant: https://github.com/ipython/ipython/pull/4762/commits/4c5aa28b94c5f8ad4872d1230ce4b6d158de6722

That was to sanitize cookies for username.. in this case I'm suggesting sanitize other cookies in browser

**Comment by minrk** [comment]
Thanks for reporting!

This is in a log function, so I wouldn't expect there to be any user-facing consequence other than noisy logs. If you're seeing errors in execution, I think something else might be involved (JupyterHub code mostly isn't involved in execution requests).

Should be fixed by #4356
